directives/work-hour-picker/work-hour-picker.component.js

```diff
@@ -0,0 +1,179 @@
+            OverNight: null,
+            OpenHour: '',
+            CloseHour: '',
+            IsSelected: false
+        },
+        {
+            Name: $translate.instant('DayOfWeekSaturday'),
+            UntranslatedName:'xxSaturday',
+            OverNight: null,
+            OpenHour: '',
+            CloseHour: ''
+        },
+        {
+            Name: $translate.instant('DayOfWeekSunday'),
+            UntranslatedName:'xxSunday',
+            OverNight: null,
+            OpenHour: '',
+            CloseHour: ''
+        }
+    ];
+
+        function toggleSelectedDay(selectedDays, day) {
+            var index = ctrl.selectedDays.indexOf(day);
+            if (index === -1) {
+                ctrl.selectedDays.push(day);
+                day.IsSelected = true;
+            }
+            else {
+                ctrl.selectedDays.splice(index,1);
+                day.IsSelected = false;
+            }
+        }
+
+        ctrl.toggleTimeFormat = function () {
+            if (ctrl.timeFormat.Meridian) {
+                ctrl.timeFormat.Format = 'HH:mm';
+                ctrl.timeFormat.Meridian = false;
+            }
+            else {
+                ctrl.timeFormat.Format = 'h:mm a';
+                ctrl.timeFormat.Meridian = true;
+            }
+        }
+
+        ctrl.closeWorkingPicker = function (form) {
+            ctrl.selectedDays = [];
+            for (var i = 0; i < ctrl.weekdays.length; i++) {
+                ctrl.weekdays[i].IsSelected = false;
+            }
+            form.closeHour = '';
+            form.openHour = '';
+            form.$invalid = false;
+            ctrl.showSummary = false;
+        }
+
+        ctrl.clearDay = function(day) {
+            day.OpenHour = '',
+            day.CloseHour = '';
+            day.OverNight = null;
+        }
+
+        ctrl.selectDay = function (day) {
+            toggleSelectedDay(ctrl.selectedDays, day)
+        }
+
+        ctrl.openPicker = function (form) {
+            ctrl.showSummary = !ctrl.showSummary;
+            if (ctrl.selectedDays.length > 0) {
+                ctrl.closeWorkingPicker(form);
+            }
+        }
+
+        function checkValidTime (form) {
+            if (!form.openHour || !form.closeHour) {
+                ctrl.errorMessage = $translate.instant('StartTimeAndEndTimeMustBeSet');
+                form.$invalid = true;
+            }
+
+            if (form.close.$invalid || form.open.$invalid) {
+                ctrl.errorMessage = $translate.instant('EnteredTimeIsInvalid');
+                form.$invalid = true;
+            }
+
+            if (form.openHour > form.closeHour && ctrl.overNight === false) {
+                ctrl.errorMessage = $translate.instant('EndTimeMustBeGreaterOrEqualToStartTime');
+                form.$invalid = true;
+            }
+            return form.$invalid;
+        }
+
+        ctrl.saveHours = function(form) {
+            if (checkValidTime(form)) {
```

Comment:
small nitpick, CheckValidTime returns true if the time is invalid, would it not make more sense if the method returned false? or if the method was named formIsInvalid()